src/main/java/hudson/plugins/throttleconcurrents/pipeline/ThrottleStep.java

```diff
@@ -0,0 +1,81 @@
+package hudson.plugins.throttleconcurrents.pipeline;
+
+import hudson.Extension;
+import hudson.model.TaskListener;
+import hudson.plugins.throttleconcurrents.ThrottleJobProperty;
+import hudson.util.FormValidation;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+public class ThrottleStep extends Step implements Serializable {
+    private String categories;
+
+    @DataBoundConstructor
+    public ThrottleStep(@Nonnull String categories) {
+        this.categories = categories;
+    }
+
+    @Nonnull
+    public String getCategories() {
+        return categories;
+    }
+
+    @Nonnull
+    public List<String> getCategoriesList() {
+        List<String> catList = new ArrayList<>();
+
+        StringTokenizer tokenizer = new StringTokenizer(categories, ",");
+        while (tokenizer.hasMoreTokens()) {
+            catList.add(tokenizer.nextToken().trim());
```

Comment:
If I understand correctly, it is going to inject empty strings. Which may become a real problem if somebody uses something like `cat_a,${myvar},cat_c` expression